addons/partner_autocomplete/static/src/js/partner_autocomplete_many2one.js

```diff
@@ -1,141 +1,72 @@
-     * Add logo in the autocomplete results if logo is provided
-     *
-     * @private
-     */
-    _modifyAutompleteRendering: function (){
-        var api = this.$input.data('ui-autocomplete');
-        // FIXME: bugfix to prevent traceback in mobile apps due to override
-        // of Many2one widget with native implementation.
-        if (!api) {
-            return;
-        }
-        api._renderItem = function(ul, item){
-            ul.addClass('o_partner_autocomplete_dropdown');
-            var $a = $('<a/>')["html"](item.label);
-            if (item.logo){
-                var $img = $('<img/>').attr('src', item.logo);
-                $a.append($img);
-            }
-
-            return $("<li></li>")
-                .data("item.autocomplete",item)
-                .append($a)
-                .appendTo(ul)
-                .addClass(item.classname);
-        };
-    },
-
-    /**
-     * @override
-     * @private
-     */
-    _renderEdit: function (){
-        this._super.apply(this, arguments);
-        this._modifyAutompleteRendering();
-    },
-
-    /**
-     * Query Autocomplete and add results to the popup
-     *
-     * @override
-     * @param search_val {string}
-     * @returns {Promise}
-     * @private
-     */
-    _searchSuggestions: function (search_val) {
-        var self = this;
-        return new Promise(function (resolve, reject) {
-            if (self._isOnline()) {
-
-                self._autocomplete(search_val).then(function (suggestions) {
-                    var choices = [];
-                    if (suggestions && suggestions.length) {
-                        _.each(suggestions, function (suggestion) {
-                            var label = '<i class="fa fa-magic text-muted"/> ';
-                            label += _.str.sprintf('%s, <span class="text-muted">%s</span>', suggestion.label, suggestion.description);
-
-                            choices.push({
-                                label: label,
-                                action: function () {
-                                    self._createPartner(suggestion);
-                                },
-                                logo: suggestion.logo,
-                                classname: 'o_partner_autocomplete_dropdown_item',
+/** @odoo-module **/
+
+import { Many2XAutocomplete } from '@web/views/fields/relational_utils';
+import { Many2OneField } from '@web/views/fields/many2one/many2one_field';
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+
+import { usePartnerAutocomplete } from "@partner_autocomplete/js/partner_autocomplete_core"
+
+export class PartnerAutoCompleteMany2XAutocomplete extends Many2XAutocomplete {
+    setup() {
+        super.setup();
+
+        this.partner_autocomplete = usePartnerAutocomplete();
+    }
+
+    validateSearchTerm(request) {
+        return request && request.length > 2;
+    }
+
+    get sources() {
```

Comment:
sources is not a getter in autocomplete. Why do you need a getter ?

Reply:
I don't really need it, but I've seen that it's how it was done in other instances where `Autocomplete` is used.
They all added a props like `sources="sources"` and then defined a `get sources()`.
Do you want that I change it to `sources="getSources()"` or something along those lines ?